src/servers/admin-api/admins/admins.controller.ts

```diff
@@ -47,49 +48,72 @@ export class AdminsController {
     }
   }
 
-  @Post('/signin')
-  async authenticate(
+  @Post('/add-book')
+  async createBook(
     @Request() request: ExpressRequest,
     @Response() response: ExpressResponse,
     @Next() next: ExpressNextFunction
   ) {
     try {
-      const account = await services.admins.signin(request.body);
-      request.session.account = account;
-      return response.status(200).json({ message: 'Login sucessful' });
+      Object.assign(request.body, { createdBy: request.session.account.id });
```

Comment:
Order matters, no point appending the `createdBy` property if the body fails the validation. 
```
const body = await validate<BookDto>(BookDto, request.body);
Object.assign(request.body, { createdBy: request.session.account.id });
const book = await services.books.registerBook(request.body);
```

The 2 lines above can also be written as 
`await services.books.registerBook({ ...request.body, createdBy: request.session.account.id })`

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax